145 saving screen settings in user defaul

semik997 opened the pull request.

semik997 committed:
Made saving the color and size of the cell in the user default

semik997 committed:
Made saving the color and size of the cell in the user default from main and favorite view

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -31,6 +31,9 @@ class FavoritesCollectionVC: UICollectionViewController, UIPopoverPresentationCo
     private let smallSize = CGSize (width: 100, height: 150)
+    private var userColor: String?
+    private var defaultColor: UIColor?
+    private var favoriteMark: SettingViewController.ViewWhitchCame = .favorite
```
Зачем хранить это здесь если можно присвоить это, по необходимости, прямо во время сегью?

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -111,26 +118,21 @@ class FavoritesCollectionVC: UICollectionViewController, UIPopoverPresentationCo
+        if segue.identifier == seguesConstant.favoriteSettings {
+            if let tvc = segue.destination as? SettingViewController {
+                tvc.delegate = self
+                tvc.viewCum = favoriteMark
```
Загугли что такое cum и исправь 😏

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -160,19 +162,45 @@ class FavoritesCollectionVC: UICollectionViewController, UIPopoverPresentationCo
+        if view == .favorite {
+            var helpingColor: UIColor?
+            if color == UIColor.white {
+                guard let chooseColor = UserDefaultManager.shared.favoriteViewSettings.color else { return }
```
Ты можешь вместо гварда проверить есть ли цвет через запятую в if let.

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -104,15 +110,10 @@ class MainCollectionVC: UICollectionViewController, UIPopoverPresentationControl
-                if let ppc = tvc.popoverPresentationController {
-                    ppc.delegate = self
-                }
+                tvc.viewCum = mainMark
```
Аналогично с тем что я писал выше.

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -133,17 +134,43 @@ class MainCollectionVC: UICollectionViewController, UIPopoverPresentationControl
+        
+        if view == .main {
+            var helpingColor: UIColor?
+            if color == UIColor.white {
```
Аналогично с тем что я писал выше.

semik997 committed:
Fixed typos

PaszhaBiceps commented on Movie Search App/CoreData/UserDefaultManager.swift:
```diff
@@ -0,0 +1,71 @@
+        
+        get {
+            if let data = defaults.value(forKey: userDefaultConst.mainSettingsKey) as? Data {
+                guard let defaultsData = try? PropertyListDecoder().decode(UserSettings.self, from: data) else { return favoriteViewSettings }
```
А почему ты в else возвращаешь favoriteViewSettings?

PaszhaBiceps commented on Movie Search App/CoreData/UserDefaultManager.swift:
```diff
@@ -0,0 +1,71 @@
+    }
+    
+    
+    func saveDefaultSettingMainView(color: String?, sizeCell: SettingViewController.ChooseSize?) {
```
Сделай это через одну функцию.

PaszhaBiceps commented on Movie Search App/TableVC/SettingViewController.swift:
```diff
@@ -43,7 +52,26 @@ class SettingViewController: UIViewController, UIColorPickerViewControllerDelega
-        mediumSizeCellButton.isSelected = true
+        switch viewCum {
+        case .main:
+            if defaultMainSize == ChooseSize.big {
```
Используй switch вместо if else здесь.

PaszhaBiceps commented on Movie Search App/TableVC/SettingViewController.swift:
```diff
@@ -43,7 +52,26 @@ class SettingViewController: UIViewController, UIColorPickerViewControllerDelega
+                smallSizeCellButton.isSelected = true
+            }
+        case .favorite:
+            if defaultFavoriteSize == ChooseSize.big {
```
Аналогично с вышесказанным.

semik997 committed:
replaced if else with enum
made changes to the color check

semik997 committed:
Added button from close setting view
Added feature change color
Change button color according to selected one
Change logic save color and size in UserDefault

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -152,27 +153,57 @@ class FavoritesCollectionVC: UICollectionViewController, UIPopoverPresentationCo
+    func updateInterface(color: UIColor?,
+                         size: SettingViewController.ChooseSize?,
+                         type: UserDefaultManager.SettingType) {
+        if type == .favoriteScreen {
```
Я не могу до конца осознать что делает этот кусок кода. Опиши его.

semik997 replied:
Transferring data and updating from the settings screen

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -152,27 +153,57 @@ class FavoritesCollectionVC: UICollectionViewController, UIPopoverPresentationCo
+            favoriteCollectionView.reloadData()
+            
+            
+            saveToUserDefault(color: helpingColor, size: size, type: type)
```
Вызывай тут сразу сохранение в дефолтсы, без использование лишней функции.

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -152,27 +153,57 @@ class FavoritesCollectionVC: UICollectionViewController, UIPopoverPresentationCo
+        
+        UserDefaultManager.shared.saveDefaultSetting(UserSettings.init(color: userStringColor, sizeCell: size),
+                                                     type: type)
+//        UserDefaultManager.shared.saveDefaultSetting(color: userStringColor, sizeCell: size, source: view)
```
Чисти ненужные комменты.

semik997 replied:
I don't have it.
I don't understand why you see them

PaszhaBiceps commented on Movie Search App/CollectionVC/FavoritesCollectionVC.swift:
```diff
@@ -152,27 +153,57 @@ class FavoritesCollectionVC: UICollectionViewController, UIPopoverPresentationCo
+    
+    func receiveFromUserDefault() {
+        
+        guard let chooseColor = UserDefaultManager.shared.getDefaultSettings(type: settingType)?.color else { return }
```
Доставай структуру один раз.

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -133,17 +133,46 @@ class MainCollectionVC: UICollectionViewController, UIPopoverPresentationControl
-// MARK: - Setting view color
+// MARK: - Setting view color and size cell
 
 extension MainCollectionVC: SettingViewControllerDelegate {
```
Подумай над тем что бы вынести эту функциональность в базовый класс.

PaszhaBiceps replied:
Либо сделать один вью контроллер для двух типов.

PaszhaBiceps commented on Movie Search App/CollectionVC/MainCollectionVC.swift:
```diff
@@ -133,17 +133,46 @@ class MainCollectionVC: UICollectionViewController, UIPopoverPresentationControl
-        } else {
-            collectionViewSpace.backgroundColor = color
-            navigationController?.navigationBar.backgroundColor = color
+    func updateInterface(color: UIColor?,
```
Аналогично с тем что я написал немного выше (к такому же куску кода).

PaszhaBiceps commented on Movie Search App/CoreData/UserDefaultManager.swift:
```diff
@@ -0,0 +1,91 @@
+    
+    private init() {}
+    
+//    var mainViewSettings: UserSettings {
```
Удаляй комментированный код если он не нужен.

PaszhaBiceps commented on Movie Search App/CoreData/UserDefaultManager.swift:
```diff
@@ -0,0 +1,91 @@
+struct UserSettings: Codable {
+    var color: String?
+    var colorButton: String?
+    var sizeCell: SettingViewController.ChooseSize?
```
Вынеси этот размер как отдельную сущность.

semik997 replied:
Не совсем понимаю что ты имеешь в виду

PaszhaBiceps commented on Movie Search App/Model+Constants/Constants.swift:
```diff
@@ -23,10 +23,19 @@ let placeholderFilm = "https://www.salonlfc.com/wp-content/uploads/2018/01/image
 struct SeguesConst {
     let showDetail = "showDetail"
-    let infoButton = "popVC"
+    let mainSettings = "mainSettings"
```
Сделай здесь тоже статические переменный и не инициализируй структуры эти нигде.

semik997 committed:
Code refactoring
Removed unnecessary functions
